Add audformat.utils.hash()

frankenjoe opened the pull request:
![image](https://user-images.githubusercontent.com/10383417/142039258-9bf7cf9f-7943-4121-801b-415e5e934eac.png)

hagenw commented:
For me it also works if I pass a Series or data frame, but it returns a slightly different value in those cases:

```python
>>> y = pd.Series([1, 2])
>>> df = pd.DataFrame([1, 2])
>>> audformat.utils.hash_index(y)
'61872891661014892'
>>> audformat.utils.hash_index(df)
'61872891661014892'
>>> audformat.utils.hash_index(y.index)
'6238072747940578789'
>>> audformat.utils.hash_index(df.index)
'6238072747940578789'
```

I see two solutions:
* raise an error if not an index is given
* add support for series and dataframe, but hash the index in those cases

hagenw commented:
I know that we usually don't raise an error if you provide the wrong input type, but the problem is that under `audformat.utils` we have a lot of functions that work the same for index, series and data frame and most likely user will expect the same for `audformat.utils.hash_index()`.
We might also rename it to `audformat.utils.hash()` and say that it is a feature that we get different results for `df` and `df.index`.

frankenjoe commented:
> We might also rename it to `audformat.utils.hash()` and say that it is a feature that we get different results for `df` and `df.index`.

Sorry, just saw your comment now. I just changed to always hashing the index. But I guess your suggestion makes sense, because you can always do `hash(obj.index)` if you only want to cache the index.

frankenjoe commented:
> > We might also rename it to `audformat.utils.hash()` and say that it is a feature that we get different results for `df` and `df.index`.
> 
> Sorry, just saw your comment now. I just changed to always hashing the index. But I guess your suggestion makes sense, because you can always do `hash(obj.index)` if you only want to cache the index.

Changed to `hash()`.

hagenw commented on audformat/core/utils.py:
```diff
@@ -285,6 +285,32 @@ def duration(
+    r"""Create hash from object.
+
+    Objects with same entries in a different order,
+    will produce the same hash string.
```
Should we add a test for this or simply trust `pandas` here?

frankenjoe replied:
This is not really related to `pandas` but the fact that we calculate the sum of the hashes we get for each row.

But since it's never a bad idea to have tests, so I added some now :)

I also rephrased to: 

![image](https://user-images.githubusercontent.com/10383417/142039796-15982637-442b-480e-80ac-43937b396759.png)

hagenw replied:
Great

hagenw commented:
> > > We might also rename it to `audformat.utils.hash()` and say that it is a feature that we get different results for `df` and `df.index`.
> > 
> > 
> > Sorry, just saw your comment now. I just changed to always hashing the index. But I guess your suggestion makes sense, because you can always do `hash(obj.index)` if you only want to cache the index.
> 
> Changed to `hash()`.

Cool, thanks.
